File: src/main/java/io/github/kingg22/api/vacunas/panama/util/NumDosisEnumConverter.java
package io.github.kingg22.api.vacunas.panama.util;

import io.github.kingg22.api.vacunas.panama.modules.vacuna.dto.NumDosisEnum;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter(autoApply = true)
public class NumDosisEnumConverter implements AttributeConverter<NumDosisEnum, String> {

    @Override
    public String convertToDatabaseColumn(final NumDosisEnum attribute) {
        if (attribute == null) {
            return null;
        }
        return attribute.getValue();
    }

    @Override
    public NumDosisEnum convertToEntityAttribute(final String dbData) {
        if (dbData == null || dbData.isBlank()) {
            return null;
        }
        return NumDosisEnum.fromValue(dbData.trim().toUpperCase());
    }
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/modules/vacuna/converter/NumDosisEnumConverter.kt
package io.github.kingg22.api.vacunas.panama.modules.vacuna.converter

import io.github.kingg22.api.vacunas.panama.modules.vacuna.dto.NumDosisEnum
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter(autoApply = true)
class NumDosisEnumConverter : AttributeConverter<NumDosisEnum?, String?> {
    override fun convertToDatabaseColumn(attribute: NumDosisEnum?): String? = attribute.let {
        it?.value ?: return null
    }

    override fun convertToEntityAttribute(dbData: String?): NumDosisEnum? = dbData.let {
        if (it.isNullOrBlank()) null else NumDosisEnum.fromValue(it.trim().uppercase())
    }
}
